frontend/src/components/camp/Picasso.vue

```diff
@@ -173,25 +174,27 @@ export default {
       return this.period().camp()
     }
   },
+  mounted () {
+    this.api.get().activities({ periodId: this.period().id })._meta.load.then(() => { this.activitiesLoading = false })
+  },
```

Comment:
Why is this request not done in CampProgram, as it is used in ActivityList as well?

Reply:
I had it there first, but it made the Picasso and ActivityList components more complicated to use (must be patched through the slot in ScheduleEntries.vue). I figured this functionality will always be useful when using the Picasso component anywhere, so I didn't want to put the responsibility on the component using the Picasso.

I do understand if you have concerns about code duplication, and I propose to address those concerns in #599.

There would be an advantage to moving it to CampProgram and SideBarProgram though: In SideBarProgram, we could only load the activities of the displayed day. But that would require another filter in the backend which was not trivial to implement: `/activities?dayId=123` needs an SQL join that calculates the `dayId` based on the `periodId` and the `periodOffset` (in minutes).

Reply:
Okay. No my "concern" was more because of [conditional rendering](https://vuejs.org/v2/guide/conditional.html#v-if-vs-v-show). If you toggle the view mode, then the reload call is made again.

Reply:
Oh, okay, I think I need to clarify about the reload. There will be no reload as long as all scheduleEntries are loaded (or loading) in the store (https://github.com/ecamp/hal-json-vuex/blob/b3eeec799e4a8328c69a385403f8ab24a5149add/src/storeValueProxy.js#L151). I.e. the code in `mounted` does run again, but `hal-json-vuex` notices that all data is already present in the store, so it doesn't request anything from the API. The `load` promise immediately resolves and the `activitiesLoading` flag is set to `false`.

You can confirm this by toggling between picasso and activity list several times and watching the network tab in the browser dev tools: After the picasso is rendered fully for the first time, there are no new network requests sent out anymore, even though the mounted hook runs every time you toggle.